### ISSUE TITLE
Migration drops IPIP tunnel address

### ISSUE THREAD

**Issue by fasaxc** [projectcalico/libcalico-go#750]
Running uplevel Felix against backlevel Typha, hit this cyclic restart in felix:
```
2017-12-18 18:52:15.612 [WARNING][449] felix.go 841: Config change requires restart key="IpInIpTunnelAddr" new="10.56.0.1" updateType="add"
```

Felix thinks the config that it loaded at start of day is different from that that typha sent in the snapshot.

## Expected Behavior

Pre/post migration data should agree.

## Current Behavior

Looks like the IPIP tunnel address is missing.

## Possible Solution

Add migration from the old config param to the calico Node BGP config resource.

## Steps to Reproduce (for bugs)

1. Start backlevel v2.6.4 cluster with typha and allow a tunnel address to be assigned
2. Upgrade calico/node to v3.0.0
3.  First calico/node to restart does the migration
4. But it gets stuck with felix cyclicly restarting.

## Your Environment

* libcalico-go version:
* Operating System and version:
* Link to your project (optional):

**Comment by fasaxc** [comment]
Should be fixed by #754.  With me to retest...

**Comment by fasaxc** [comment]
Doesn't appear to have been fixed with the latest calico/node master:
```
$ kubectl exec -n kube-system  calico-node-2c2vz versions
Defaulting container name to calico-node.
Use 'kubectl describe pod/calico-node-2c2vz' to see all of the containers in this pod.
confd v1.0.0-beta1-8-g621d697
BIRD version 1.6.3
libnetwork-plugin: v1.1.2-rc1
calico-bgp-daemon: v0.2.2

Felix:
Version:            3.0.0-beta1-69-gfa15f177
Full git commit ID: fa15f1776244d1db9e30fcd99f51b8b99be509c8
Build date:         2017-12-19T08:00:34+0000
```
I believe the problem is that the fix in #754 only addresses etcd, but I was testing the GKE scenario, which uses KDD.

**Comment by fasaxc** [comment]
So I think the problem is that, in v2.6.x, if a node has a podCIDR, then it has an implicit IpInIpTunnelAddress but in v3.0.0, we only consider it to have the implicit address if the node _also_ has other BGP configuration.
